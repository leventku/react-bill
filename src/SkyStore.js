import React from 'react';

import BillGroup from './BillGroup';

function TestStore(props) {
  const prepareItems = (data) => {
    let allItems = [];
    Object.keys(data).filter( key => key !== 'total').forEach( category => {
      data[category].forEach(item => {
        allItems.push(Object.assign({type: category}, item))
      })
    })
    return allItems;
  }

  const renderItem = (item, i) => (
    <li className="list-group-item justify-content-between" key={i}>
      <div>
        <span className="badge badge-primary" style={{width: 100}}>{item.type}</span>
        <span className="name" style={{padding: '0 10px'}}>{item.title}</span>
      </div>
      <span className="cost">Â£{item.cost.toFixed(2)}</span>
    </li>
  )

  return (
    <BillGroup 
      title="Sky Store"
      items={prepareItems(props.data)}
      itemRenderer={renderItem}
      total={props.data.total}
    />
  )
}

export default TestStore;