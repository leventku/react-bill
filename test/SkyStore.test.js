import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SkyStore from '../src/SkyStore';

function setupComponent() {
  const props = {
    data: {
      "rentals": [
        { "title": "50 Shades of Grey", "cost": 4.99 }
      ],
      "buyAndKeep": [
        { "title": "That's what she said", "cost": 9.99 },
        { "title": "Brokeback mountain", "cost": 9.99 }
      ],
      "total": 24.97
    }
  }

  const wrapper = shallow(<SkyStore {...props} />)

  return  {
    props,
    wrapper
  }
}

describe('SkyStore component', () => {
  it('should render', () => {
    const {wrapper} = setupComponent();
    expect(wrapper.exists()).toBe(true);
  })

  it('should render its items', () => {
    const {wrapper} = setupComponent();
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})