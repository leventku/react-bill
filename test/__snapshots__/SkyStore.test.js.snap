// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkyStore component should render its items 1`] = `
<BillGroup
  itemRenderer={[Function]}
  items={
    Array [
      Object {
        "cost": 4.99,
        "title": "50 Shades of Grey",
        "type": "rentals",
      },
      Object {
        "cost": 9.99,
        "title": "That's what she said",
        "type": "buy and keep",
      },
      Object {
        "cost": 9.99,
        "title": "Brokeback mountain",
        "type": "buy and keep",
      },
    ]
  }
  title="Sky Store"
  total={24.97}
/>
`;
